### PR TITLE
Update common-errors.md

### PR DESCRIPTION
**Pull request by lavinir1**
Adding details on throttling and recommended behaviors in error cases

### CODE REVIEW

**Comment by lavinir1** [comment]
@microsoft-github-policy-service agree [company="{microsoft}"]

**Comment by lavinir1** [comment]
@microsoft-github-policy-service agree

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 1]
```diff
@@ -20,14 +20,16 @@ ms.custom: api
 search.appverid: met150
 ---
 
-# Common REST API error codes
-
+# Handling REST API Errors
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
# Handling REST API errors 
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 2]
```diff
@@ -20,14 +20,16 @@ ms.custom: api
...
-# Common REST API error codes
-
+# Handling REST API Errors 
 
 
 [!INCLUDE [Microsoft 365 Defender rebranding](../../includes/microsoft-defender.md)]
 
+HTTP error responses are divided to 2 categories:
+* Client error (400-code level) – client sent an invalid request or not according to defeintions.
+* Server error (500-level) – server failed temporarly to fulfill the request or server error occured. reseding the HTTP reqeust might help
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
HTTP error responses are divided into two categories:
* Client error (400-code level) – the client sent an invalid request or the request is not in accordance with definitions.
* Server error (500-level) – the server temporarily failed to fulfill the request or a server error occurred. Try sending the HTTP request again.
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 3]
```diff
@@ -59,8 +61,16 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
 DisallowedOperation|Forbidden (403)|{the disallowed operation and the reason}.
 NotFound|Not Found (404)|General Not Found error message.
 ResourceNotFound|Not Found (404)|Resource {the requested resource} was not found.
-InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
 TooManyRequests|Too Many Requests (429)|Response will represent reaching quota limit either by number of requests or by CPU.
+InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
+
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
InternalServerError|Internal Server Error (500)|(No error message, retry the operation.)
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 4]
```diff
@@ -59,8 +61,16 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
...
+
+## Throttling   
+Clients may receive 'Too Many Requests error (429)' when the number of HTTP requests in a given time frame exceed the allowed number of calls per API
+
+The client should delay resubmitting farther HTTPS requests in a way that comply with the rate limitations.
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
You should delay resubmitting further HTTPS requests and then submit them in a way that complies with the rate limitations. 
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 5]
```diff
@@ -59,8 +61,16 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
...
 TooManyRequests|Too Many Requests (429)|Response will represent reaching quota limit either by number of requests or by CPU.
+InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
+
+
+## Throttling   
+Clients may receive 'Too Many Requests error (429)' when the number of HTTP requests in a given time frame exceed the allowed number of calls per API
+
+The client should delay resubmitting farther HTTPS requests in a way that comply with the rate limitations. 
+
+Ignoring 429 response or trying to resubmit HTTP reqeusts in a shorter time frame will cause the the calculated requests rate to stay high for longer time frame in which the server will keep responding with 429 error code
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
Ignoring the 429 response or trying to resubmit HTTP requests in a shorter time frame will cause the calculated request rate to stay high for a longer time, during which the server will keep responding with a 429 error code.
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 6]
```diff
@@ -59,8 +61,16 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
 DisallowedOperation|Forbidden (403)|{the disallowed operation and the reason}.
 NotFound|Not Found (404)|General Not Found error message.
 ResourceNotFound|Not Found (404)|Resource {the requested resource} was not found.
-InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
 TooManyRequests|Too Many Requests (429)|Response will represent reaching quota limit either by number of requests or by CPU.
+InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
+
+
+## Throttling   
+Clients may receive 'Too Many Requests error (429)' when the number of HTTP requests in a given time frame exceed the allowed number of calls per API
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
You may receive a 'Too Many Requests error (429)' when the number of HTTP requests in a given time frame exceeds the allowed number of calls per API.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review @lavinir1, kindly commit the suggested changes. Many thanks.

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
@lavinir1, thank you for committing my suggested changes, but you seem to have missed one. Could you also commit that suggestion, please?

**Comment by lavinir1** [comment]
> @lavinir1, thank you for committing my suggested changes, but you seem to have missed one. Could you also commit that suggestion, please?

Why to change the name of the actor from 'client' to 'you' ?

**Comment by JohanFreelancer9** [comment]
> > @lavinir1, thank you for committing my suggested changes, but you seem to have missed one. Could you also commit that suggestion, please?
> 
> Why to change the name of the actor from 'client' to 'you' ?

Thank you for your question. In accordance with the Microsoft Style Guide, we don't use "client" to refer to a person in Microsoft docs. Furthermore, it is consistent with the previous sentence where you used "you". The remainder of the changes to this line is to correct a spelling and grammatical error.

**Review by JohanFreelancer9 — APPROVED** [review]

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 7]
```diff
@@ -59,8 +60,15 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
...
 NotFound|Not Found (404)|General Not Found error message.
 ResourceNotFound|Not Found (404)|Resource {the requested resource} was not found.
-InternalServerError|Internal Server Error (500)|(No error message, retry the operation)
 TooManyRequests|Too Many Requests (429)|Response will represent reaching quota limit either by number of requests or by CPU.
+InternalServerError|Internal Server Error (500)|(No error message, retry the operation.)
+
+## Throttling   
+The HTTP client may receive a 'Too Many Requests error (429)' when the number of HTTP requests in a given time frame exceeds the allowed number of calls per API.
+
+The HTTP client should delay resubmitting further HTTPS requests and then submit them in a way that complies with the rate limitations. A Retry-After in the response header indicating how long to wait (in seconds) before making a new request
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md:69]:
```suggestion
The HTTP client should delay resubmitting further HTTPS requests and then submit them in a way that complies with the rate limitations. A Retry-After in the response header indicates how long to wait (in seconds) before making a new request.
```

File: microsoft-365/security/defender-endpoint/common-errors.md [hunk 8]
```diff
@@ -59,8 +60,15 @@ DisabledFeature|Forbidden (403)|Tenant feature is not enabled.
...
+
+Ignoring the 429 response or trying to resubmit HTTP requests in a shorter time frame will cause return of 429 error code.
```
Comment (JohanFreelancer9) [inline microsoft-365/security/defender-endpoint/common-errors.md]:
```suggestion
Ignoring the 429 response or trying to resubmit HTTP requests in a shorter time frame will cause a return of the 429 error code.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review.

**Review by JohanFreelancer9 — APPROVED** [review]

**Pull request merged**